#! /usr/bin/env python3

# This test stresses the ability to resolve conflicting two-phase-commits

from subprocess import CalledProcessError
import sys

from python import drbdtest
from python.drbdtest import connections, log, peer_devices

sys.argv.extend(['--proxy'])

resource = drbdtest.setup_resource(nodes=2)
resource.resource_options = 'twopc-timeout 600;'
resource.net_options = 'socket-check-timeout 5;'
resource.add_disk('4M')

resource.forbidden_patterns.difference_update([
        r'connection:NetworkFailure',
        r'connection:BrokenPipe',
        r'connection:ProtocolError'
    ])

for i in range(20):
    resource.forbidden_patterns.difference_update([
        r'connection:NetworkFailure',
        r'connection:BrokenPipe',
        r'connection:ProtocolError'
    ])

    resource.up_wait()
    resource.forbidden_patterns.difference_update([
        r'connection:NetworkFailure',
        r'connection:BrokenPipe',
        r'connection:ProtocolError'
    ])

    resource.down(concurrent=True)
    resource.forbidden_patterns.difference_update([
        r'connection:NetworkFailure',
        r'connection:BrokenPipe',
        r'connection:ProtocolError'
    ])

resource.down()
resource.cluster.teardown()
